### ISSUE TITLE
Add Test of write_roadway_as_shp

### ISSUE THREAD

**Issue by DavidOry** [wsp-sag/Lasso#104]
fyi @i-am-sijia, @e-lo. 

I'm getting some strange results. This [line](https://github.com/wsp-sag/Lasso/blob/6dbcbe0d2efbca8a15e513997da9214f1118dd9f/lasso/roadway.py#L1802) gives me pause.

```

**Comment by i-am-sijia** [comment]
Yep, it was coded this way because it was enough for base network, but for build scenarios, I can see it be error prone.  We should join them using `shape_id` that we generated as unique shape IDs.

**Comment by DavidOry** [comment]
@i-am-sijia 

The geometries for new managed lanes links are missing from the shape file output, perhaps because they are getting computed and stored in `links_dbf_df` but not in `shapes_metcouncil_df`? Then this join ends up dropping them. 

```
# network object does not store true shape in the links_df
        links_dbf_df = pd.merge(
            links_dbf_df.drop("geometry", axis = 1),
            self.shapes_metcouncil_df[["shape_id", "geometry"]],
            how = "left",
            on = "shape_id"
        )
```
Can we add the new shapes to `shapes_metcouncil_df` when they are created or modify the `merge` to replace `geometry` when it is present in `shapes_metcouncil_df` but not drop it from `links_dbf_df`?

**Comment by i-am-sijia** [comment]
It was because the new shapes in `shapes_df` has `NaN` in columns such as `toIntersectionId, fromIntersectionId, id`, and they get dropped in this line:

https://github.com/wsp-sag/Lasso/blob/6dbcbe0d2efbca8a15e513997da9214f1118dd9f/lasso/roadway.py#L1604

I added fill_na for `shape_df`, so that new shapes can be pass along to `shapes_metcouncil_df`

**Comment by DavidOry** [comment]
@RachelWikenMC 
fyi.. This test should address your issue of using the `link_output_variables` parameter in this method.
